nucliadb/nucliadb/search/api/v1/resource/ask.py

```diff
@@ -0,0 +1,143 @@
+from typing import Optional, Union
+
+from fastapi import Header, Request, Response
+from fastapi_versioning import version
+from starlette.responses import StreamingResponse
+
+from nucliadb.common import datamanagers
+from nucliadb.models.responses import HTTPClientError
+from nucliadb.search.api.v1.router import KB_PREFIX, RESOURCE_SLUG_PREFIX, api
+from nucliadb_models.resource import NucliaDBRoles
+from nucliadb_models.search import AskRequest, NucliaDBClientType
+from nucliadb_utils.authentication import requires
+
+from ..ask import create_ask_response
+
+
+@api.post(
+    f"/{KB_PREFIX}/{{kbid}}/resource/{{rid}}/ask",
+    status_code=200,
+    summary="Ask with a resource (by id)",
+    description="Ask questions to a resource",
+    tags=["Search"],
+    response_model=None,
+)
+@requires(NucliaDBRoles.READER)
+@version(1)
+async def resource_ask_endpoint_by_uuid(
+    request: Request,
+    kbid: str,
+    rid: str,
+    item: AskRequest,
+    x_ndb_client: NucliaDBClientType = Header(NucliaDBClientType.API),
+    x_nucliadb_user: str = Header(""),
+    x_forwarded_for: str = Header(""),
+    x_synchronous: bool = Header(
+        False,
+        description="When set to true, outputs response as JSON in a non-streaming way. "
+        "This is slower and requires waiting for entire answer to be ready.",
+    ),
+) -> Union[StreamingResponse, HTTPClientError, Response]:
+    return await resource_ask_endpoint(
+        request,
+        kbid,
+        item,
+        x_ndb_client,
+        x_nucliadb_user,
+        x_forwarded_for,
+        x_synchronous,
+        resource_id=rid,
+    )
+
+
+@api.post(
+    f"/{KB_PREFIX}/{{kbid}}/{RESOURCE_SLUG_PREFIX}/{{slug}}/ask",
+    status_code=200,
+    summary="Ask with a resource (by slug)",
+    description="Ask with a resource",
+    tags=["Search"],
+    response_model=None,
+)
+@requires(NucliaDBRoles.READER)
+@version(1)
+async def resource_ask_endpoint_by_slug(
+    request: Request,
+    kbid: str,
+    slug: str,
+    item: AskRequest,
+    x_ndb_client: NucliaDBClientType = Header(NucliaDBClientType.API),
+    x_nucliadb_user: str = Header(""),
+    x_forwarded_for: str = Header(""),
+    x_synchronous: bool = Header(
+        False,
+        description="When set to true, outputs response as JSON in a non-streaming way. "
+        "This is slower and requires waiting for entire answer to be ready.",
+    ),
+) -> Union[StreamingResponse, HTTPClientError, Response]:
+    return await resource_ask_endpoint(
+        request,
+        kbid,
+        item,
+        x_ndb_client,
+        x_nucliadb_user,
+        x_forwarded_for,
+        x_synchronous,
+        resource_slug=slug,
+    )
+
+
+async def resource_ask_endpoint(
+    request: Request,
+    kbid: str,
+    item: AskRequest,
+    x_ndb_client: NucliaDBClientType,
+    x_nucliadb_user: str,
+    x_forwarded_for: str,
+    x_synchronous: bool,
+    resource_id: Optional[str] = None,
+    resource_slug: Optional[str] = None,
```

Comment:
To avoid id/slug conditionals, I'd prefer you to always expect a resource id and convert the slug to id in the specific function. I fixed other writer methods to follow this pattern